src/cockpit/channel.py

```diff
@@ -506,13 +521,9 @@ def do_close(self) -> None:
     def do_ping(self, message: JsonObject) -> None:
         self.receive_queue.put_nowait(message)
 
-    def do_data(self, data: bytes) -> None:
-        if not isinstance(data, bytes):
-            # this will persist past this callback, so make sure we take our
-            # own copy, in case this was a memoryview into a bytearray.
-            data = bytes(data)
-
+    def do_data(self, data: bytes) -> bool:
         self.receive_queue.put_nowait(data)
+        return True  # we will send the 'ack' later
```

Comment:
That "later" is in the `read()` method? Comment could make this more precise.

Reply:
Sure.